protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -107,7 +107,8 @@ private T parseFrom(InputStream stream) throws InvalidProtocolBufferException {
    */
   public static <T extends Message> Metadata.Key<T> keyForProto(final T instance) {
     return Metadata.Key.of(
-        instance.getDescriptorForType().getFullName() + Metadata.BINARY_HEADER_SUFFIX,
+        instance.getDescriptorForType().getFullName().replace('.', '-')
```

Comment:
Yeah, this was a legitimate breakage, but one we didn't realize earlier. It's to be used by users, and it has to be compatible between implementations.